### PR TITLE
Conflictvisualizer Map Page Integration

### PR DESCRIPTION
**Pull request by jacob6838**
# PR Details
This PR integrates the jpo-conflictvisualizer map page into the cvmanager. This change allows intersection data, collected with the conflictmonitor, to be visualized in the intersection map page of the cvmanager. This is the first step in integrating the functionality of the cvmanager and conflictvisualizer tools together!

## Description

This change adds a new intersection map page, which combines the Conflictvisualizer map page with the existing RsuMap intersection view.

## How Has This Been Tested?

These changes have been tested by visualizing MAP/SPAT/BSM data on the map, and comparing the displays with the original jpo-conflictvisualizer. 
These changes were tested running locally in docker. These changes require a jpo-Tconflictvisualizer api to be running, and to be connected to the cvmanager keycloak server (and cvmanager keycloak realm). This API also requires the jpo-conflictmonitor and jpo-geojsonconverter to be running, so that there is data available. 
These changes do not affect the API or the Keycloak server. 

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [x] My changes require new environment variables:
  - [x] I have updated the docker-compose, K8s YAML, and all dependent deployment configuration files.
- [x] My changes require updates to the documentation:
  - [ ] I have updated the documentation accordingly.
- [x] My changes require updates and/or additions to the unit tests:
  - [ ] I have modified/added tests to cover my changes.
- [x] All existing tests pass.

### CODE REVIEW

**Comment by jacob6838** [comment]
I created this PR as a draft, because I still have to complete a few tasks: Adding unit tests for the jpo-conflictvisualizer map page, and fixing merge conflicts

**Comment by drewjj** [comment]
We should definitely add the required services to a docker-compose and provide information/links to deploying this so this can be easily replicated. I want to test this but I want to make sure I am doing this correctly since I have never run the conflict visualizer locally before. New users of the cv-manager will be even more clueless.

File: webapp/README.md [hunk 1]
```diff
@@ -106,6 +112,14 @@ Re-factoring RSU manager to utilize Redux Toolkit for state management
...
      - WzdxMap.js - read and load WZDx data
 
+## ConflictVisualizer Integration
+
+The usdot [jpo-conflictvisualizer](https://github.com/usdot-jpo-ode/jpo-conflictvisualizer) is a tool that can be used to visualize the conflicts between Basic Safety Messages (BSMs), Signal Phase and Timing (SPaT) messages, and MAP messages. The CV Manager Web Application has been integrated with the ConflictVisualizer tool to allow users to view data directly from a [jpo-conflictmonitor](https://github.com/usdot-jpo-ode/jpo-conflictmonitor) instance. This integration currently requires an additional jpo-conflictvisualizer api to be deployed alongside the jpo-cvmanager api. This allows the webapp to make authenticated requests to the jpo-conflictvisualizer api to retrieve the conflict monitor data.
+
+### Changes Made
+
+These changes were tested running locally in docker. These changes require a jpo-conflictvisualizer api to be running, and to be connected to the cvmanager keycloak server (and cvmanager keycloak realm). This API also requires the jpo-conflictmonitor and jpo-geojsonconverter to be running, so that there is data available. Once
+
```
Comment (dmccoystephenson) [inline webapp/README.md]: Looks like this sentence ends abruptly; Is there supposed to be more after "Once"?

File: webapp/public/icons/traffic-light-icon-green-1.svg
Comment (dmccoystephenson) [inline webapp/public/icons/traffic-light-icon-green-1.svg:1]: Is there a reason for having this copy of the `traffic-light-icon-green.svg` file?

File: webapp/public/icons/traffic-light-icon-red-1.svg
Comment (dmccoystephenson) [inline webapp/public/icons/traffic-light-icon-red-1.svg:1]: Is there a reason to have this copy of `traffic-light-icon-red.svg`?

File: webapp/src/apis/api-helper-cviz.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      body: body
+        ? localHeaders['Content-Type'] === 'application/x-www-form-urlencoded'
+          ? (body as string)
+          : JSON.stringify(body)
+        : undefined,
+      mode: 'cors',
+      signal: controller?.signal,
+    }
+
+    console.debug('MAKING REQUEST TO ' + url + ' WITH OPTIONS', options)
```
Comment (dmccoystephenson) [inline webapp/src/apis/api-helper-cviz.ts]: Should a space be inserted between `WITH OPTIONS` and the options variable here for readability purposes?

File: webapp/src/apis/configuration-param-api.ts [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+// /config/intersection/lane_direction_of_travel/heading_tolerance
+// /config/intersection/general/v2x_message_processing_frequency
+// /config/intersection/general/spat_minimum_10_second_reception
+// /config/intersection/general/spat_maximum_10_second_reception
+// /config/intersection/general/message_storage_period
+// /config/intersection/general/map_minimum_10_second_reception
+// /config/intersection/general/map_maximum_10_second_reception
+// /config/intersection/connection_of_travel/minimum_number_of_events
+// /config/intersection/connection_of_travel/look_back_period
+// ]
```
Comment (dmccoystephenson) [inline webapp/src/apis/configuration-param-api.ts]: What's the purpose of these comments? Should they be removed?

File: webapp/src/apis/events-api.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+interface Item {
+  label: string
+  value: string
+}
+
+const EVENT_TYPES: Item[] = [
+  { label: 'ConnectionOfTravelEvent', value: 'connection_of_travel' },
+  { label: 'IntersectionReferenceAlignmentEvent', value: 'intersection_reference_alignment' },
+  { label: 'LaneDirectionOfTravelEvent', value: 'lane_direction_of_travel' },
+  //   { label: "ProcessingTimePeriod", value: "processing_time_period" },
```
Comment (dmccoystephenson) [inline webapp/src/apis/events-api.ts]: Why is this commented out?

File: webapp/src/apis/events-api.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    { latest = false }: { latest?: boolean } = {}
+  ): Promise<MessageMonitor.Event[]> {
+    const queryParams = {
+      intersection_id: intersectionId.toString(),
+      road_regulator_id: roadRegulatorId.toString(),
+      start_time_utc_millis: startTime.getTime().toString(),
+      end_time_utc_millis: endTime.getTime().toString(),
+      latest: latest.toString(),
+    }
+    // if (roadRegulatorId) queryParams["road_regulator_id"] = roadRegulatorId;
```
Comment (dmccoystephenson) [inline webapp/src/apis/events-api.ts]: Should this and other commented-out code in this file be removed?

File: webapp/src/components/map/control-panel.tsx [hunk 6]
```diff
@@ -0,0 +1,751 @@
...
+
+const Accordion = styled((props: AccordionProps) => <MuiAccordion disableGutters elevation={0} square {...props} />)(
+  ({ theme }) => ({
+    //border: `1px solid ${theme.palette.divider}`,
+    // '&:not(:last-child)': {
+    //   borderBottom: 0,
+    // },
+    // '&:before': {
+    //   display: 'none',
+    // },
```
Comment (dmccoystephenson) [inline webapp/src/components/map/control-panel.tsx]: Should this be commented out?

File: webapp/src/components/map/map-legend.tsx [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+          <div style={{ height: 20, width: 20, backgroundColor: value, marginLeft: 2 }}></div>
+        </div>
+        <p>|</p>
+      </>
+    )
+  }
+
+  const travelConnectionColorsList: JSX.Element[] = []
+  for (const [key, value] of Object.entries(travelConnectionColors)) {
+    const heightFactor = 20 / 2 / value[1].reduce((partialSum, a) => partialSum + a, 0) // 20 is the height of the colored bar, 2 is the number of sets of colors
```
Comment (dmccoystephenson) [inline webapp/src/components/map/map-legend.tsx]: This comment could be replaced with two additional constants with descriptive names.

File: webapp/src/components/map/utilities/message-utils.ts [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+  connectingLanes: MapFeatureCollection
+) => {
+  const features: any[] = []
+  const markerCollection = {
+    type: 'FeatureCollection' as 'FeatureCollection',
+    features: features,
+  }
+  switch (notification.notificationType) {
+    case 'ConnectionOfTravelNotification':
+      // TODO: Re-add once more notification data is available
```
Comment (dmccoystephenson) [inline webapp/src/components/map/utilities/message-utils.ts]: Should we log to the console that this is unimplemented at the moment?

File: webapp/src/icons/map/traffic-light-icon-red.svg
Comment (dmccoystephenson) [inline webapp/src/icons/map/traffic-light-icon-red.svg:1]: I am seeing the following on a few of these SVG files; is this indicative of a problem?

image

File: webapp/src/models/MessageMonitorNotification.d.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
+type MessageMonitorNotification =
+  | IntersectionReferenceAlignmentNotification
+  | SignalGroupAlignmentNotification
+  // | SpatMinimumDataNotification //
+  // | MapMinimumDataNotification //
+  | SpatBroadcastRateNotification
+  | MapBroadcastRateNotification
+  // | StopLineStopNotification //
+  // | LaneDirectionOfTravelAssessmentNotification
```
Comment (dmccoystephenson) [inline webapp/src/models/MessageMonitorNotification.d.ts]: Why are these commented out?

File: webapp/src/models/convert_java_to_ts.py
Comment (dmccoystephenson) [inline webapp/src/models/convert_java_to_ts.py:1]: Why is this in the models directory?

File: webapp/src/models/imports/ProcessedMap.d.ts [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+type MapSource = 'RSU' | 'V2X' | 'MMITSS' | 'unknown'
+
+// enum MapSource {
+//     RSU,
+//     V2X,
+//     MMITSS,
+//     unknown,
+// };
```
Comment (dmccoystephenson) [inline webapp/src/models/imports/ProcessedMap.d.ts]: Is this still needed?

File: webapp/src/models/imports/README.md [hunk 11]
```diff
@@ -0,0 +1 @@
+# Manually Generated External Imports
```
Comment (dmccoystephenson) [inline webapp/src/models/imports/README.md:1]: Should this file have more content?

File: webapp/src/models/jpo-conflictmonitor/config/ConfigDataClass.d.ts
Comment (dmccoystephenson) [inline webapp/src/models/jpo-conflictmonitor/config/ConfigDataClass.d.ts:1]: Why is this empty?

File: webapp/src/components/map/map-slice.tsx [hunk 12]
```diff
@@ -0,0 +1,1576 @@
...
+            }, numRestartsLocal * 2000)
+          )
+        )
+      } else {
+        cleanUpLiveStreaming()
+      }
+    }
+
+    client.onStompError = (frame) => {
+      // TODO: Consider restarting connection on error
```
Comment (dmccoystephenson) [inline webapp/src/components/map/map-slice.tsx]: Is this something that should be implemented in this PR?

File: webapp/src/components/map/map-slice.tsx [hunk 13]
```diff
@@ -0,0 +1,1576 @@
...
+      //     if (elem.type === 'srmTx') {
+      //       localSrmCount += 1
+      //     } else {
+      //       localSsmCount += 1
+      //     }
+      //   }
+      // }
+      // state.value.srmCount = localSrmCount
+      // state.value.srmSsmCount = localSsmCount
+      // state.value.srmMsgList = localMsgList
```
Comment (dmccoystephenson) [inline webapp/src/components/map/map-slice.tsx]: Should this be commented out?

File: webapp/src/generalSlices/rsuSlice.ts [hunk 14]
```diff
@@ -1,567 +1,568 @@
...
+        state.requestOut = false
+        state.value.messageLoading = false
+      })
+      .addCase(updateGeoMsgData.pending, (state) => {
+        console.debug('updateGeoMsgData pending')
+        state.loading = true
+        state.value.addGeoMsgPoint = false
+        // Removed 1 day limitation for new mongo deployment
+        // state.value.geoMsgDateError =
+        //   new Date(state.value.geoMsgEnd).getTime() - new Date(state.value.geoMsgStart).getTime() > 86400000
```
Comment (dmccoystephenson) [inline webapp/src/generalSlices/rsuSlice.ts]: Are these comments still relevant?

**Review by dmccoystephenson — CHANGES_REQUESTED** [review]
I reviewed the changed files and left some questions & comments. Please note that I have not yet run the unit tests with these changes.

File: webapp/src/apis/api-helper-cviz.ts (hunk 2 above)
Comment (jacob6838) [reply webapp/src/apis/api-helper-cviz.ts]: In console log statements, arguments separated by a comma are automatically spaced out from the other arguments, and printed next to them. So this would print: "MAKING REQUEST TO URL WITH OPTIONS {...options}"

File: webapp/src/apis/events-api.ts (hunk 5 above)
Comment (jacob6838) [reply webapp/src/apis/events-api.ts]: old comment, removed!

File: webapp/src/apis/events-api.ts (hunk 4 above)
Comment (jacob6838) [reply webapp/src/apis/events-api.ts]: old comment, removed!

File: webapp/src/components/map/control-panel.tsx (hunk 6 above)
Comment (jacob6838) [reply webapp/src/components/map/control-panel.tsx]: old comment, removed!

File: webapp/src/components/map/utilities/message-utils.ts (hunk 8 above)
Comment (jacob6838) [reply webapp/src/components/map/utilities/message-utils.ts]: done!

File: webapp/src/components/map/map-legend.tsx (hunk 7 above)
Comment (jacob6838) [reply webapp/src/components/map/map-legend.tsx]: done!

File: webapp/src/models/convert_java_to_ts.py
Comment (jacob6838) [reply webapp/src/models/convert_java_to_ts.py:1]: This script converts Java classes to .d.ts classes. I added a README.md to describe this process!

File: webapp/src/models/imports/ProcessedMap.d.ts (hunk 10 above)
Comment (jacob6838) [reply webapp/src/models/imports/ProcessedMap.d.ts]: Removed!

File: webapp/src/models/jpo-conflictmonitor/config/ConfigDataClass.d.ts
Comment (jacob6838) [reply webapp/src/models/jpo-conflictmonitor/config/ConfigDataClass.d.ts:1]: Removed!

File: webapp/src/components/map/map-slice.tsx (hunk 12 above)
Comment (jacob6838) [reply webapp/src/components/map/map-slice.tsx]: Removed the comment. This callback firing does not signify that the websocket is dead, so should not restart, but can e helpful for debugging other errors

File: webapp/src/generalSlices/rsuSlice.ts (hunk 14 above)
Comment (jacob6838) [reply webapp/src/generalSlices/rsuSlice.ts]: I am not sure... my git merge from develop seems to have screwed up my git lens. Let me check with the author on this!

File: webapp/src/components/map/map-slice.tsx (hunk 13 above)
Comment (jacob6838) [reply webapp/src/components/map/map-slice.tsx]: This is not currently used, but might be on the future... Implemented!

File: webapp/src/models/imports/README.md (hunk 11 above)
Comment (jacob6838) [reply webapp/src/models/imports/README.md:1]: Added content here!

File: webapp/src/models/MessageMonitorNotification.d.ts (hunk 9 above)
Comment (jacob6838) [reply webapp/src/models/MessageMonitorNotification.d.ts]: This file was completely unused, removed!

File: webapp/public/icons/traffic-light-icon-green-1.svg
Comment (jacob6838) [reply webapp/public/icons/traffic-light-icon-green-1.svg:1]: These were versions of the light icons with dimmer non-lit colors. These are no longer necessary, removed!

File: webapp/public/icons/traffic-light-icon-red-1.svg
Comment (jacob6838) [reply webapp/public/icons/traffic-light-icon-red-1.svg:1]: Removed, see above!

File: webapp/src/apis/configuration-param-api.ts (hunk 3 above)
Comment (jacob6838) [reply webapp/src/apis/configuration-param-api.ts]: Removed!

File: webapp/src/icons/map/traffic-light-icon-red.svg
Comment (jacob6838) [reply webapp/src/icons/map/traffic-light-icon-red.svg:1]: Not sure what happened to these, but they weren't being used, so I removed them

File: webapp/README.md (hunk 1 above)
Comment (jacob6838) [reply webapp/README.md]: Updated this documentation!

File: webapp/src/generalSlices/rsuSlice.ts (hunk 14 above)
Comment (jacob6838) [reply webapp/src/generalSlices/rsuSlice.ts]: Checked with Michael, removed!

File: docker/connect/Dockerfile
Comment (dmccoystephenson) [inline docker/connect/Dockerfile:1]: Should this dockerfile have an entrypoint?

File: docker/mongo/Dockerfile
Comment (dmccoystephenson) [inline docker/mongo/Dockerfile:1]: Same here, does this dockerfile need an entrypoint?

File: docker/mongo/a_init_replicas.js [hunk 15]
```diff
@@ -0,0 +1,33 @@
+
+/*
+
+This script is the first of two scripts responsible for setting up mongoDB on initial boot. 
+These scripts should be copied to the /docker-entrypoint-initdb.d/ directory within the docker image;
+the docker image will then execute these scripts automatically when the database is first created. 
+This script and is partner are prefixed with the letters a and b respectively to ensure they are run
```
Comment (dmccoystephenson) [inline docker/mongo/a_init_replicas.js]: Should this be "This script and **its** partner"?

File: docker/mongo/mongo_initialize.sh
Comment (dmccoystephenson) [inline docker/mongo/mongo_initialize.sh:1]: Is this file still needed? Looks like the contents are commented out

**Review by dmccoystephenson — CHANGES_REQUESTED** [review]
Looking good, just a few more comments.

Also, can you please merge the `develop` branch into `conflictvisualizer-map-page` to ensure the 'files changed' section of the PR is as accurate as possible?

File: docker/mongo/Dockerfile
Comment (jacob6838) [reply docker/mongo/Dockerfile:1]: This was not used, removed!

File: docker/mongo/a_init_replicas.js (hunk 15 above)
Comment (jacob6838) [reply docker/mongo/a_init_replicas.js]: updated!

File: docker/mongo/mongo_initialize.sh
Comment (jacob6838) [reply docker/mongo/mongo_initialize.sh:1]: Removed!

File: docker/connect/Dockerfile
Comment (jacob6838) [reply docker/connect/Dockerfile:1]: This appears to be handled by the base image confluentinc/cp-kafka-connect

File: conflictmonitor/docker/mongo/b_create_indexes.js [hunk 16]
```diff
@@ -0,0 +1,619 @@
...
+do {
+  try {
+    missing_collection_count = 0
+    const collectionNames = db.getCollectionNames()
+    for (collection of collections) {
+      // Create Collection if it doesn't exist
+      let created = false
+      if (!collectionNames.includes(collection.name)) {
+        created = createCollection(collection)
+        // created = true;
```
Comment (dmccoystephenson) [inline conflictmonitor/docker/mongo/b_create_indexes.js]: This comment can probably be removed

File: sample.env [hunk 17]
```diff
@@ -77,13 +77,6 @@ CSM_EMAILS_TO_SEND_TO=
 CSM_TARGET_SMTP_SERVER_ADDRESS=
 CSM_TARGET_SMTP_SERVER_PORT=
 
-# Email configuration for sending firmware-manager failure emails
-SMTP_EMAIL=
-SMTP_USERNAME=
-SMTP_PASSWORD=
-FW_EMAIL_RECIPIENTS=
-SMTP_SERVER_IP=
-
```
Comment (dmccoystephenson) [inline sample.env:105]: Why was this removed?

**Review by dmccoystephenson — CHANGES_REQUESTED** [review]

File: sample.env (hunk 17 above)
Comment (jacob6838) [reply sample.env:105]: good catch, sorry! I replaced it with another sample... Fixed this now

File: conflictmonitor/docker/mongo/b_create_indexes.js (hunk 16 above)
Comment (jacob6838) [reply conflictmonitor/docker/mongo/b_create_indexes.js]: Removed!

**Review by dmccoystephenson — CHANGES_REQUESTED** [review]
The python unit tests pass for me, but I am running into test failures with the webapp tests:

image

I am seeing the following error predominantly:
> TypeError: Color is not a function

Using npm version 10.7.0

**Comment by jacob6838** [comment]
> The python unit tests pass for me, but I am running into test failures with the webapp tests:
> 
> image
> 
> I am seeing the following error predominantly:
> 
> > TypeError: Color is not a function
> 
> Using npm version 10.7.0

Just pushed a commit to resolve this issue!

**Review by dmccoystephenson — APPROVED** [review]
Code changes look good & the unit tests (both services & webapp) pass!

Note: I have not spun up the CV Manager application with these changes.

**Review by drewjj — APPROVED** [review]
Overall, I am approving this because everything behaves well, tests pass, and the code generally looks good. I couldn't possibly go through every line but I looked into the new intersection slice, new API calls, and some of the new components.

I do think we should consider renaming the map components. This can be pretty confusing with the main Map page not having much to do with each other. I understand the components are called this because they visualize CV Map messages but maybe we should call them cvmap or v2xmap components instead.

I know this is generally a first step for the Conflict Monitor integration so there is more room for slimming things down. This PR does come across as a major addition to the technical effort that users will need to get this running and for future developers to be able to properly work on it. I think the docker-compose additions helps with this a lot. I am still trying to grapple with what we could do for developers when working on the cv manager web app. Maybe we could try to refactor names. Not sure.

File: README.md [hunk 18]
```diff
@@ -48,6 +48,29 @@ The JPO CV Manager was originally developed for the Google Cloud Platform and a
...
+- The CV Manager webapp has been integrated with the ConflictVisualizer tool to allow users to view data directly from a jpo-conflictmonitor instance. This integration currently requires an additional jpo-conflictvisualizer api to be deployed alongside the jpo-cvmanager api. This allows the webapp to make authenticated requests to the jpo-conflictvisualizer api to retrieve the conflict monitor data.
+- All of the components required for the ConflictVisualizer integration are available in the usdot-jpo-ode GitHub repositories:
+  - [jpo-conflictvisualizer (api)](https://github.com/usdot-jpo-ode/jpo-conflictvisualizer/tree/cvmgr-cimms-integration/api)
+  - [jpo-conflictmonitor](https://github.com/usdot-jpo-ode/jpo-conflictmonitor)
+  - [jpo-geojsonconverter](https://github.com/usdot-jpo-ode/jpo-geojsonconverter)
+  - [jpo-ode](https://github.com/usdot-jpo-ode/jpo-ode)
+
+#### Running the ConflictMonitor and ConflictVisualizer API
+
+Use the release-compose.yml to pull images for the jpo-conflictviualizer-api-cvmanager, jpo-conflictmonitor, jpo-geojsonconverter, and jpo-ode.
```
Comment (payneBrandon) [inline README.md]: I don't see a `release-compose.yml` file here?

Comment (drewjj) [reply README.md]: Good point, he removed it when he made the additions to the docker-compose instead.

File: conflictmonitor/README.md [hunk 19]
```diff
@@ -0,0 +1,7 @@
+# JPO-Conflictmonitor Components
```
Comment (payneBrandon) [inline conflictmonitor/README.md:1]: Is this directory being used in this project? Or just a convenience for when you spin up an associated conflict monitor?

File: conflictmonitor/docker/mongo/b_create_indexes.js [hunk 20]
```diff
@@ -0,0 +1,618 @@
+// Create indexes on all collections
+
+/*
+This is the second script responsible for configuring mongoDB automatically on startup.
+This script is responsible for creating users, creating collections, adding indexes, and configuring TTLs
+For more information see the header in a_init_replicas.js
+*/
+
+console.log('')
```
Comment (payneBrandon) [inline conflictmonitor/docker/mongo/b_create_indexes.js]: Needed?

File: docker-compose.yml [hunk 21]
```diff
@@ -71,6 +71,10 @@ services:
         MAPBOX_INIT_LATITUDE: ${MAPBOX_INIT_LATITUDE}
         MAPBOX_INIT_LONGITUDE: ${MAPBOX_INIT_LONGITUDE}
         MAPBOX_INIT_ZOOM: ${MAPBOX_INIT_ZOOM}
+        CVIZ_MAPBOX_TOKEN: ${CVIZ_MAPBOX_TOKEN}
```
Comment (payneBrandon) [inline docker-compose.yml]: Does this need to exist separate from the MAPBOX_TOKEN above?

File: README.md (hunk 18 above)
Comment (jacob6838) [reply README.md]: Updated the docs, and added a command to run the docker compose up without the conflcitmonitor components!

File: conflictmonitor/README.md (hunk 19 above)
Comment (jacob6838) [reply conflictmonitor/README.md:1]: These are necessary for running the conflictmonitor and conflictvisualizer together, as this contains the kafka connect config. Without using a submodule, I am not sure how else to include these files. They are sufficiently complex that it would be difficult to run a successful local development system without them

File: conflictmonitor/docker/mongo/b_create_indexes.js (hunk 20 above)
Comment (jacob6838) [reply conflictmonitor/docker/mongo/b_create_indexes.js]: Is this referring to the entire file, or the prints?

Comment (dmccoystephenson) [reply conflictmonitor/docker/mongo/b_create_indexes.js]: > Is this referring to the entire file, or the prints?

Probably the log statement

Comment (drewjj) [reply conflictmonitor/docker/mongo/b_create_indexes.js]: Definitely the empty log console.log('')

File: conflictmonitor/README.md (hunk 19 above)
Comment (payneBrandon) [reply conflictmonitor/README.md:1]: Let's expand on this README a bit to explain how these files are used, and why they are necessary. Currently we just mention there's "useful configuration files" but this makes it seem like they are necessary to run the application properly. We could punt and add this as a backlog item as well.

File: conflictmonitor/docker/mongo/b_create_indexes.js (hunk 20 above)
Comment (payneBrandon) [reply conflictmonitor/docker/mongo/b_create_indexes.js]: Yes, just the empty log statement. Not sure what it was added in here.

File: docker-compose-webapp-deployment.yml [hunk 22]
```diff
@@ -17,6 +17,10 @@ services:
         MAPBOX_INIT_LATITUDE: ${MAPBOX_INIT_LATITUDE}
         MAPBOX_INIT_LONGITUDE: ${MAPBOX_INIT_LONGITUDE}
         MAPBOX_INIT_ZOOM: ${MAPBOX_INIT_ZOOM}
+        CVIZ_MAPBOX_TOKEN: ${CVIZ_MAPBOX_TOKEN}
```
Comment (payneBrandon) [inline docker-compose-webapp-deployment.yml]: Same as previous, is this needed to be a separate mapbox token than the one we already inject above?

File: docker-compose.yml (hunk 21 above)
Comment (jacob6838) [reply docker-compose.yml]: Removed this requirement

File: docker-compose-webapp-deployment.yml (hunk 22 above)
Comment (jacob6838) [reply docker-compose-webapp-deployment.yml]: Removed this requirement

File: conflictmonitor/docker/mongo/b_create_indexes.js (hunk 20 above)
Comment (jacob6838) [reply conflictmonitor/docker/mongo/b_create_indexes.js]: Removed!

File: conflictmonitor/datasets/uturn_export/OdeBsmJson.json
Comment (payneBrandon) [inline conflictmonitor/datasets/uturn_export/OdeBsmJson.json:1]: On these JSON files, are they required to be fully spaced JSON? If it's just an initial datasource maybe we could reduce size by compressing these files?

File: webapp/src/apis/assessments-api.ts [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+  async getAssessments(
+    token: string,
+    eventType: string,
+    intersectionId: number,
+    roadRegulatorId: number,
+    startTime?: Date,
+    endTime?: Date
+  ): Promise<Assessment[]> {
+    const queryParams: Record<string, string> = {}
+    // queryParams["road_regulator_id"] = road_regulator_id;
```
Comment (payneBrandon) [inline webapp/src/apis/assessments-api.ts]: Remove?

File: webapp/src/components/css/RsuMapView.css [hunk 24]
```diff
@@ -9,46 +9,8 @@
   padding-left: 30px;
   padding-right: 30px;
   font-size: 1em;
-  background: white;
+  /* background: white; */
```
Comment (payneBrandon) [inline webapp/src/components/css/RsuMapView.css]: There's a couple of these commented values we could probably clean up in here

**Review by payneBrandon — CHANGES_REQUESTED** [review]
Just a couple outstanding comments

**Review by payneBrandon — APPROVED** [review]
lgtm!
